File: c#/EmployeeManagement/Models/FTInstructor.cs
namespace EmployeeManagement.Models
{
    // Full-time instructor employee with an office assignment.
    public class FTInstructor : FTEmployee
    {
        public string OfficeId { get; }

        public FTInstructor(int id, string firstName, string lastName, DateOnly dateOfBirth, string position, Departments department, double salary, string officeId)
            : base(id, firstName, lastName, dateOfBirth, position, department, salary)
        {
            OfficeId = !string.IsNullOrWhiteSpace(officeId)
                ? officeId
                : throw new System.ArgumentException("Office id cannot be null or empty.", nameof(officeId));
        }

        public override void ReportToManager()
        {
            Console.WriteLine($"FT Instructor {FirstName} {LastName} reports to the manager from office {OfficeId}.");
        }

        public override string ToString()
        {
            return base.ToString() + $", OfficeId={OfficeId}";
        }
    }
}

File: Java/FTInstructor.java
public class FTInstructor extends FTEmployee {

    private String office_id;

    public FTInstructor(
        int id,
        String first_name,
        String last_name,
        String dob,
        String position,
        Departments department,
        double salary,
        String office_id
    ) throws InvalidDataException {
        super(id, first_name, last_name, dob, position, department, salary);
        setOfficeId(office_id);
    }

    @Override
    public void reportToManager() {
        System.out.println("Instructor " + getFirstName() + " " + getLastName() + " reports to Manager.");
    }

    public String getOfficeId() {
        return office_id;
    }

    public void setOfficeId(String office_id) throws InvalidDataException {
        if (office_id == null || office_id.isBlank()) {
            throw new InvalidDataException("Office id is required.");
        }
        this.office_id = office_id;
    }
}
